ov/id_pools.go

```diff
@@ -0,0 +1,338 @@
+	var (
+		uri = "/rest/id-pools/"
+		ids StartStopFragments
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType + "/generate"
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return ids, err
+	}
+
+	log.Debugf("GenerateIds %s", data)
+	if err := json.Unmarshal(data, &ids); err != nil {
+		return ids, err
+	}
+	return ids, nil
+}
+
+// Enables or disables the pool
+func (c *OVClient) UpdatePoolType(idPool IdPool, poolType string) (IdPool, error) {
+
+	log.Infof("Initializing update of pool type for %s.", poolType)
+	var (
+		uri      = "/rest/id-pools/"
+		t        *Task
+		response IdPool
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+
+	log.Debugf("REST : %s \n %+v\n", uri, idPool)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.PUT, uri, idPool)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting update request: %s", err)
+		return response, err
+	}
+
+	log.Debugf("Response update Id Pool %s", data)
+
+	dataResponse := data
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	if err := json.Unmarshal([]byte(dataResponse), &response); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	return response, nil
+}
+
+// Allocates one or more IDs from a pool.
+func (c *OVClient) Allocator(allocateIds UpdateAllocatorList, poolType string) (UpdateAllocatorList, error) {
+
+	log.Infof("Initializing update of pool type for %s.", poolType)
+	var (
+		uri      = "/rest/id-pools/"
+		t        *Task
+		response UpdateAllocatorList
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType + "/allocator"
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+
+	log.Debugf("REST : %s \n %+v\n", uri, allocateIds)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.PUT, uri, allocateIds)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting update request: %s", err)
+		return response, err
+	}
+
+	log.Debugf("Response allocate Ids %s", data)
+
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	if err := json.Unmarshal(data, &response); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
```

Comment:
Is this task un-marshal or response un-marshal..?

Reply:
we unmarshall data to &response.